trinity/components/builtin/preferred_node/component.py

```diff
@@ -0,0 +1,117 @@
+        return DEFAULT_MAX_PEERS
+
+    @property
+    def preferred_nodes(self) -> List[NodeAPI]:
+        return self._boot_info.args.preferred_nodes
+
+    async def do_run(self, event_bus: EndpointAPI) -> None:
+        service = PreferredNodeService(event_bus, self.preferred_nodes, self.max_peers)
+        async with background_trio_service(service) as manager:
+            await manager.wait_finished()
+
+
+class PreferredNodeService(Service):
+    preferred_node_recycle_time: int = 300
+    pool_availability_check_period: int = 5
+    _preferred_node_tracker: Dict[NodeAPI, float] = None
+
+    def __init__(
+        self, event_bus: EndpointAPI, preferred_nodes: List[NodeAPI], max_peers: int
+    ) -> None:
+        self.event_bus = event_bus
+        self.max_peers = max_peers
+        self.preferred_nodes = preferred_nodes
+        self._preferred_node_tracker = collections.defaultdict(lambda: 0)
+        self.logger = get_logger('trinity.components.preferred_node.PreferredNodeService')
+
+    async def run(self) -> None:
+        while self.manager.is_running:
+            await self.periodically_check_pool_for_peer_availability()
+
+    async def periodically_check_pool_for_peer_availability(self) -> None:
+        """
+        Check pool for available peer slot, and trigger command to connect
+        to a preferred node if there's room.
+        """
+        async for _ in trio_utils.every(self.pool_availability_check_period):
+            await self.event_bus.wait_until_any_endpoint_subscribed_to(GetConnectedPeersRequest)
+            response = await self.event_bus.request(GetConnectedPeersRequest())
+
+            if len(response.peers) <= self.max_peers:
+                connected_peers = [peer.session.remote for peer in response.peers]
+                await self.connect_pool_to_preferred_peer(connected_peers)
+
+    async def connect_pool_to_preferred_peer(self, connected_peers: List[NodeAPI]) -> None:
+        """
+        Send a command to peer pool to connect to a preferred peer, if such a
+        connection attempt has not been made within preferred_node_recycle_time.
+        """
+        try:
+            random_preferred_node = self._get_random_preferred_node()
+        except NoEligibleNodes:
+            pass
+        else:
+            if random_preferred_node not in connected_peers:
+                await self.event_bus.broadcast(
+                    ConnectToNodeCommand(random_preferred_node),
+                    TO_NETWORKING_BROADCAST_CONFIG,
+                )
+
+    @to_tuple
+    def _get_eligible_preferred_nodes(self) -> Iterator[NodeAPI]:
+        """
+        Return nodes from the preferred_nodes which have not been used within
+        the last preferred_node_recycle_time
+        """
+        for node in self.preferred_nodes:
+            last_used = self._preferred_node_tracker[node]
+            if time.time() - last_used > self.preferred_node_recycle_time:
```

Comment:
```suggestion
            if time.monotonic() - last_used > self.preferred_node_recycle_time:
```

Monotonic is better here to handle daylight savings and other clock shifts.